src/snowflake/cli/plugins/spcs/image_repository/manager.py

```diff
@@ -15,38 +21,43 @@ def get_schema(self):
     def get_role(self):
         return self._conn.role
 
-    def get_repository_url_list(self, repo_name: str) -> SnowflakeCursor:
-        role = self.get_role()
+    def get_repository_row(self, repo_name: str) -> Dict:
+        if not is_valid_identifier(repo_name):
+            raise ValueError(
+                f"repo_name '{repo_name}' is not a valid Snowflake identifier"
+            )
         database = self.get_database()
         schema = self.get_schema()
+        name_is_quoted = is_valid_quoted_identifier(repo_name)
 
-        registry_query = f"""
-            use role {role};
-            use database {database};
-            use schema {schema};
-            show image repositories like '{repo_name}';
-            """
-
-        return self._execute_query(registry_query)
-
-    def get_repository_url(self, repo_name):
-        database = self.get_database()
-        schema = self.get_schema()
+        # if repo_name is a quoted identifier, strip quotes before using as pattern
+        repo_name_raw = repo_name[1:-1] if name_is_quoted else repo_name.upper()
```

Comment:
We have `unquote_identifier` already that can help with this -- it doesn't change `%` or `_` though like your `escape_like_pattern` function does. By the way, why is `_` needed to be escaped?

Reply:
Perhaps we can create an `identifier_for_like_clause` utility that combines `unquote_identifier` and the `%` escaping!

Reply:
when used in a LIKE pattern, _ gets interpreted as a single character wildcard

Reply:
Good idea to combine the two.

One confusing thing is that the LIKE in SHOW LIKE ... and regular LIKE (https://docs.snowflake.com/en/sql-reference/functions/like) behave differently. SHOW LIKE is case insensitive and LIKE is not. We may have to specify something like `identifier_for_show_like_clause`.

Reply:
One more note is that in this case unquote_identifier actually doesn't work as it changes the identifier fully to uppercase. Later on in the function I need the case-sensitive version of the unquoted repo_name in order to compare against a list of candidate results while `unquote_identifier` casts it to all uppercase.

For example, if our repo name is "test_REPO", there may be both TEST_REPO and "test_REPO" existing in the same schema, both of which would be returned by "SHOW IMAGE REPOSITORIES LIKE 'test_REPO'. Then, I need the case sensitivity to determine which one to actually return.